lib/validators/options_processor/options/no_empty.js

```diff
@@ -18,15 +20,17 @@ class NoEmpty {
       }
     }
 
-    let isMergeable
+    let isMergeable = false
 
     const DEFAULT_SUCCESS_MESSAGE = `The ${validatorContext.name} is not empty`
     if (!description) description = `The ${validatorContext.name} can't be empty`
 
     if (typeof input === 'string') {
       isMergeable = input.trim().length !== 0
     } else if (Array.isArray(input)) {
-      isMergeable = input.length !== 0
+      isMergeable = input[0] != null
+    } else if (input == null || _.isUndefined(input)) {
```

Comment:
rather than have it like this, I think it would be cleaner to have 
```
} else if (!(input == null || _.isUndefined(input)) {
throw new Error(UNKNOWN_INPUT_TYPE_ERROR)
} 
```

and remove the last `else`

Reply:
@shine2lay it needs to have the `else if` block so it `null` or `undefined` will be `false` instead of throwing error.

Reply:
you can easily avoid negative conditionals (not (A or B)) if you take the inverse of `or`, instead of `A or B` you can do `not A and not B` that way you can avoid that negation in the front. I personally have no issue either way